gradle-suppressible-error-prone/src/test/groovy/com/palantir/gradle/suppressibleerrorprone/SuppressibleErrorPronePluginIntegrationTest.groovy

```diff
@@ -0,0 +1,531 @@
+
+        // language=Java
+        appJava.text == '''
+            package app;
+            public final class App {
+                @SuppressWarnings("for-rollout:ArrayToString")
+                public final String field = new int[3].toString();
+
+                @SuppressWarnings("for-rollout:ArrayToString")
+                public void method() {
+                    System.out.println(new int[3].toString());
+                }
+
+                public void variables() {
+                    @SuppressWarnings("for-rollout:ArrayToString")
+                    String variable = new int[3].toString();
+                    System.out.println(variable);
+                }
+                
+                @SuppressWarnings("for-rollout:ArrayToString")
+                public static class SomeClass {
+                    static {
+                        System.out.println(new int[3].toString());
+                    }
+                }
+            }
+        '''.stripIndent(true)
+    }
+
+    def 'supports errorprone checks that match on a larger element than they report errors on'() {
+        // The UnusedVariable check implements CompilationUnitTreeMatcher, so will start with a whole
+        // CompilationUnitTree and then narrows down to the specific variable declaration that is unused.
+        // This trips up the "naive" suppression logic, which looks at where the visitor has got to rather
+        // than where the diagnostic description was produced.
+
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public void variables() {
+                    String variable;
+                }
+            }
+        '''.stripIndent(true)
+
+        when:
+        runTasksSuccessfully('compileJava', '-PerrorProneSuppressStage1')
+        runTasksSuccessfully('compileJava', '-PerrorProneSuppressStage2')
+
+        then:
+        runTasksSuccessfully('compileJava')
+
+        // language=Java
+        appJava.text == '''
+            package app;
+            public final class App {
+                public void variables() {
+                    @SuppressWarnings("for-rollout:UnusedVariable")
+                    String variable;
+                }
+            }
+        '''.stripIndent(true)
+    }
+
+    def 'can disable errorprone using property'() {
+        when:
+        // language=Java
+        writeJavaSourceFile '''
+            package app;
+            public final class App {
+                public static void main(String[] args) {
+                    System.out.println(new int[3].toString());
+                }
+            }
+        '''.stripIndent(true)
+
+        then:
+        runTasksSuccessfully('compileJava', '-PerrorProneDisable')
+        runTasksSuccessfully('compileJava', '-Pcom.palantir.baseline-error-prone.disable')
+    }
+
+    def 'should be able to refactor near usages of deprecated methods'() {
```

Comment:
what actually goes wrong/what are you fixing here, I'm not aware

Reply:
Added a comment